src/aws-cpp-sdk-core/source/client/ClientConfiguration.cpp

```diff
@@ -222,14 +230,6 @@ void setLegacyClientConfigurationParameters(ClientConfiguration& clientConfig)
             client->SetEndpoint(ec2MetadataServiceEndpoint);
         }
     }
-
-    clientConfig.appId = clientConfig.LoadConfigFromEnvOrProfile(
```

Comment:
this actually didnt work before because if region was present in a [env var or profile configuration](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/source/client/ClientConfiguration.cpp#L195-L212) it would short circuit and not evaluate this

Reply:
this a not nice side-effect within `LoadConfigFromEnvOrProfile`, I'd rather fix the method as well.

Reply:
not sure if I 100 percent understand but it doesnt have to do with `LoadConfigFromEnvOrProfile` it had to do with the [return calls in client config](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/source/client/ClientConfiguration.cpp#L195-L212). but yeah i should fix this function moving all region code to its own function.